satpy/readers/viirs_edr.py

```diff
@@ -0,0 +1,309 @@
+    scene = satpy.Scene(filenames, reader='viirs_edr')
+    scene.load(['smoke_concentration'])
+
+.. note::
+
+    Multiple products contain datasets with the same name! For example, both the cloud mask
+    and aerosol detection files contain a cloud mask, but these are not identical.
+    For clarity, the aerosol file cloudmask is named `cloud_mask_adp` in this reader.
+
+"""
+
+
+import logging
+
+import numpy as np
+import xarray as xr
+
+from satpy import DataID
+from satpy.readers.file_handlers import BaseFileHandler
+from satpy.utils import get_chunk_size_limit
+
+LOG = logging.getLogger(__name__)
+M_COLS = 3200
+
+
+class VIIRSJRRFileHandler(BaseFileHandler):
+    """NetCDF4 reader for VIIRS Active Fires."""
+
+    def __init__(self, filename, filename_info, filetype_info):
+        """Initialize the geo filehandler."""
+        super(VIIRSJRRFileHandler, self).__init__(filename, filename_info,
+                                                  filetype_info)
+        # use entire scans as chunks
+        row_chunks_m = max(get_chunk_size_limit() // 4 // M_COLS, 1)  # 32-bit floats
+        row_chunks_i = row_chunks_m * 2
+        self.nc = xr.open_dataset(self.filename,
+                                  decode_cf=True,
+                                  mask_and_scale=True,
+                                  chunks={
+                                      'Columns': -1,
+                                      'Rows': row_chunks_m,
+                                      'Along_Scan_375m': -1,
+                                      'Along_Track_375m': row_chunks_i,
+                                      'Along_Scan_750m': -1,
+                                      'Along_Track_750m': row_chunks_m,
+                                  })
+        if 'Columns' in self.nc.dims:
+            self.nc = self.nc.rename({'Columns': 'x', 'Rows': 'y'})
+        elif 'Along_Track_375m' in self.nc.dims:
+            self.nc = self.nc.rename({'Along_Scan_375m': 'x', 'Along_Track_375m': 'y'})
+            self.nc = self.nc.rename({'Along_Scan_750m': 'x', 'Along_Track_750m': 'y'})
+
+        # For some reason, no 'standard_name' is defined in some netCDF files, so
+        # here we manually make the definitions.
+        if 'Latitude' in self.nc:
+            self.nc['Latitude'].attrs.update({'standard_name': 'latitude'})
+        if 'Longitude' in self.nc:
+            self.nc['Longitude'].attrs.update({'standard_name': 'longitude'})
+
+        self.algorithm_version = filename_info['platform_shortname']
+        self.sensor_name = 'viirs'
+
+    def rows_per_scans(self, data_arr: xr.DataArray) -> int:
+        """Get number of array rows per instrument scan based on data resolution."""
+        return 16 if data_arr.shape[1] == M_COLS else 32
+
+    def get_dataset(self, dataset_id: DataID, info: dict) -> xr.DataArray:
+        """Get the dataset."""
+        data_arr = self.nc[info['file_key']]
+        data_arr = self._mask_invalid(data_arr, info)
```

Comment:
Should this be made optional? Some users may wish to access the unmasked data.

Reply:
For the NDVI/EVI special case with QF checks, I think it could be optional since that's very complex and opinionated. Otherwise, I'm not sure I agree. This is checking fill values and valid ranges. Why would a user need to have that? The L2 products don't encode anything with different fill values as far as I know...I think, right?

Reply:
Sorry, I meant to comment on the *other* `_mask_invalid`, the one for the NDVI/EVI. If we could make that one optional then that'd be great. This one, that I actually comment on, is fine.

Reply:
This is done now. I had a very non-creative moment and went with `filter_veg` as the kwarg. Other ideas?